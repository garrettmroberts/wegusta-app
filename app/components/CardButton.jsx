import React from 'react';
import { Pressable, View, StyleSheet, Text } from 'react-native';
import { Feather } from '@expo/vector-icons';
import { colors } from '../config';

const CardButton = props => {
  return (
    <Pressable { ...props }
      style={ ({pressed}) => [
        {
          opacity: pressed ? 0.8 : 1
        },
        styles.cardButton
      ]}
    >
      <View style={ styles.cardButton }>
        <Feather name="feather" size={40} color={colors.white} />
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  cardButton: {
    height: 80,
    width: 80,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 40,
    margin: 16,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4
  }
});

export default CardButton;
