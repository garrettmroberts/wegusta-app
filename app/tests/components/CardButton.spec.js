import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Text, Pressable } from 'react-native';
import CardButton from '../../components/CardButton';

describe('Button Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CardButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes extra props to Pressable component', () => {
    const button = shallow(<CardButton prop_1="true" />);
    const pressable = button.find(Pressable);
    expect(pressable.props().prop_1).toEqual('true');
  });
});
