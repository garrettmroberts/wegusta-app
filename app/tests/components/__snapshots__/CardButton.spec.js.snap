// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 40,
        "elevation": 4,
        "height": 80,
        "justifyContent": "center",
        "margin": 16,
        "shadowColor": "#1C262D",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "width": 80,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 40,
        "elevation": 4,
        "height": 80,
        "justifyContent": "center",
        "margin": 16,
        "shadowColor": "#1C262D",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "width": 80,
      }
    }
  />
</View>
`;
